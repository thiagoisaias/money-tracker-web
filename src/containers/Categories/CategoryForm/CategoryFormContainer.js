import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import {
  clearCategoryToEdit,
  createCategory,
  updateCategory,
  setCategoryToEdit
} from "../../../store/actions/categories/categories";

import CategoryForm from "../../../components/Categories/CategoryForm/CategoryForm";

export class CategoryFormContainer extends Component {
  submitData = categoryData => {
    const { history, isLoading, onCreateCategory } = this.props;

    if (isLoading) {
      return;
    }
    onCreateCategory(categoryData, history);
  };

  render() {
    const { categoryToEdit, error, isLoading, match } = this.props;
    return (
      <CategoryForm
        {...categoryToEdit}
        error={error}
        isLoading={isLoading}
        match={match}
        submitData={this.submitData}
      />
    );
  }

  componentDidMount() {}
}

CategoryFormContainer.propTypes = {
  categoryToEdit: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired
  }),
  error: PropTypes.string,
  history: PropTypes.object.isRequired,
  isLoading: PropTypes.bool.isRequired,
  match: PropTypes.object.isRequired,
  onClearCategoryToEdit: PropTypes.func.isRequired,
  onCreateCategory: PropTypes.func.isRequired,
  onUpdateCategory: PropTypes.func.isRequired,
  onSetCategoryToEdit: PropTypes.func.isRequired
};

const mapStateToProps = state => {
  return {
    categoryToEdit: state.categories.categoryToEdit,
    error: state.categories.error,
    isLoading: state.categories.isLoading
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onClearCategoryToEdit: () => {},
    onCreateCategory: (categoryData, history) => {
      dispatch(createCategory(categoryData, history));
    },

    onUpdateCategory: () => {},
    onSetCategoryToEdit: () => {}
  };
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(CategoryFormContainer)
);
