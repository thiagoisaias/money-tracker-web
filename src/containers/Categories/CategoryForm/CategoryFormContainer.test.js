import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { CategoryFormContainer } from "./CategoryFormContainer";

describe("CategoryFormContainer", () => {
  const mockProps = {
    categoryToEdit: {
      id: 1,
      name: "Lorem",
      color: "red"
    },
    error: null,
    history: {},
    isLoading: false,
    match: {},
    onClearCategoryToEdit: jest.fn(),
    onCreateCategory: jest.fn(),
    onUpdateCategory: jest.fn(),
    onSetCategoryToEdit: jest.fn()
  };
  const component = shallow(<CategoryFormContainer {...mockProps} />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
