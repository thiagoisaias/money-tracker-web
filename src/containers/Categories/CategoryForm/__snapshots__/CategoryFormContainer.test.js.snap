// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryFormContainer renders properly 1`] = `
<Component
  color="red"
  error={null}
  id={1}
  isLoading={false}
  match={Object {}}
  name="Lorem"
  submitData={[Function]}
/>
`;
